dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,79 +234,124 @@ std::wstring Microsoft::Windows::AppNotifications::Helpers::SetDisplayNameBasedO
     return displayName;
 }
 
-void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring const& appId, std::wstring const& clsid)
+// Placeholder
+HRESULT RetrieveAssetsFromProcess(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
 {
-    wil::unique_hkey hKey;
-    // subKey: \Software\Classes\AppUserModelId\{AppGUID}
-    std::wstring subKey{ c_appIdentifierPath + appId };
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
 
-    THROW_IF_WIN32_ERROR(RegCreateKeyEx(
-        HKEY_CURRENT_USER,
-        subKey.c_str(),
-        0,
-        nullptr /* lpClass */,
-        REG_OPTION_NON_VOLATILE,
-        KEY_ALL_ACCESS,
-        nullptr /* lpSecurityAttributes */,
-        &hKey,
-        nullptr /* lpdwDisposition */));
+    return E_NOTIMPL;
+}
 
-    // Retrieve the display name
-    std::wstring displayName{};
-    displayName = SetDisplayNameBasedOnProcessName();
+// Do nothing. This is just a placeholder while the UDK is ingested with the proper API.
+HRESULT ToastNotifications_RetrieveAssets_Stub(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
+
+    return E_NOTIMPL;
+}
+
+HRESULT RetrieveAssetsFromWindow(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
 
-    // Retrieve the icon
-    std::wstring iconFilePath{};
+    wil::unique_cotaskmem_string localDisplayName;
+    wil::unique_cotaskmem_string localIconFilePath;
 
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+    HWND hWindow = GetForegroundWindow();
```

Comment:
HWND: Looks like a leak and you may need to use the wil construct instead.

Reply:
HWNDs don't really "leak" per se.